packages/astro/src/core/compile/style.ts

```diff
@@ -1,27 +1,32 @@
+import fs from 'fs';
 import type { TransformOptions } from '@astrojs/compiler';
-import type { SourceMapInput } from 'rollup';
-import type { TransformStyle } from './types';
+import { preprocessCSS, ResolvedConfig } from 'vite';
+import { AstroErrorCodes } from '../errors/codes.js';
+import { CSSError } from '../errors/errors.js';
+import { positionAt } from '../errors/index.js';
 
-type PreprocessStyle = TransformOptions['preprocessStyle'];
-
-export function createStylePreprocessor(
-	transformStyle: TransformStyle,
-	cssDeps: Set<string>,
-	errors: Error[]
-): PreprocessStyle {
-	const preprocessStyle: PreprocessStyle = async (value: string, attrs: Record<string, string>) => {
+export function createStylePreprocessor({
+	filename,
+	viteConfig,
+	cssDeps,
+	cssTransformErrors,
+}: {
+	filename: string;
+	viteConfig: ResolvedConfig;
+	cssDeps: Set<string>;
+	cssTransformErrors: Error[];
+}): TransformOptions['preprocessStyle'] {
+	return async (content, attrs) => {
 		const lang = `.${attrs?.lang || 'css'}`.toLowerCase();
-
+		const id = `${filename}?astro&type=style&lang${lang}`;
 		try {
-			const result = await transformStyle(value, lang);
-
-			if (!result) return null as any; // TODO: add type in compiler to fix "any"
+			const result = await preprocessCSS(content, id, viteConfig);
 
-			for (const dep of result.deps) {
+			result.deps?.forEach((dep) => {
```

Comment:
nit: any reason to refactor to a `forEach` here?

Reply:
`result.deps` is now nullable so `for const` doesn't work too nice with it without wrapping an `if` condition. Thought a `?.forEach` would be cleaner 😄